src/compressor/brotli/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+# brotli
+
+set(brotli_sources 
+  CompressionPluginBrotli.cc
+  BrotliCompressor.cc
+)
+include(ExternalProject)
+ExternalProject_Add(brotli_ext
+  DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+  GIT_REPOSITORY "https://github.com/google/brotli.git"
+  GIT_TAG "master"
+  SOURCE_DIR ${CMAKE_BINARY_DIR}/src/brotli
+  CONFIGURE_COMMAND ./configure-cmake --disable-debug 
+  INSTALL_COMMAND ""
+  BUILD_COMMAND $(MAKE)
+  BUILD_IN_SOURCE 1
+  INSTALL_COMMAND "")
+
+ExternalProject_Add_Step(brotli_ext forcebuild
+  DEPENDEES configure
+  DEPENDERS build
+  COMMAND "true"
+  ALWAYS 1)
+
+set(bortli_libs brotli::common brotli::dec brotli::enc)
```

Comment:
nit, could avoid the string manipulation, and dedup the `brotli::` prefix, see https://github.com/ceph/ceph/blob/28c79fac6eb580dbb3a1bbb05c77a95ab75c277c/cmake/modules/BuildDPDK.cmake#L88